python/cuml/test/test_tsne.py

```diff
@@ -22,99 +22,101 @@
 from cuml.test.utils import stress_param
 from cuml.neighbors import NearestNeighbors as cuKNN
 
-from sklearn.neighbors import NearestNeighbors as skKNN
 from sklearn.datasets import make_blobs
-from sklearn.manifold.t_sne import trustworthiness
+from cuml.metrics import trustworthiness
 from sklearn import datasets
 
-import cuml.common.logger as logger
 
+DEFAULT_N_NEIGHBORS = 25
 
-dataset_names = ['digits', 'boston', 'iris', 'breast_cancer',
-                 'diabetes']
+test_datasets = {"digits": datasets.load_digits(),
+                 "boston": datasets.load_boston(),
+                 "cancer": datasets.load_breast_cancer(),
+                 "diabetes": datasets.load_diabetes()}
```

Comment:
This looks much better since it will only load each dataset once. My only concern with this is it will load all of the datasets when the module is loaded, even if no test is run. This will become an issue when running parallel tests since each worker would load a copy of all datasets, even if they arent running it. A better system would be to use a simple factory fixture to load and cache the datasets on demand. For example (I tested this and it works well):

```python
test_datasets = {"digits": datasets.load_digits,
                 "boston": datasets.load_boston,
                 "cancer": datasets.load_breast_cancer,
                 "diabetes": datasets.load_diabetes}


@pytest.fixture(scope="module")
def dataset():

    loaded_datasets = {}

    def _dataset(name):

        if (name not in loaded_datasets):
            loaded_datasets[name] = test_datasets[name]()
            
        return loaded_datasets[name]

    return _dataset
```
And then change the test function to:
```python
@pytest.mark.parametrize('name', test_datasets.keys())
@pytest.mark.parametrize('type_knn_graph', ['cuml', 'sklearn'])
@pytest.mark.parametrize('method', ['fft', 'barnes_hut'])
def test_tsne_knn_graph_used(dataset, name, type_knn_graph, method):

    X = dataset(name).data
```
Note: For some of the later tests that use a single dataset, you can load that with `dataset("digits").data` which will use the cached value.